api/src/utils/csv-utils/csv-config-validation.ts

```diff
@@ -30,18 +33,36 @@ export const validateCSVWorksheet = <StaticHeaderType extends Uppercase<string>>
     return { errors: errors, rows: [] };
   }
 
-  // Iterate over each cell in the worksheet and validate + set cell values
-  forEachCSVCell(worksheet, config, (params, headerConfig) => {
-    // Validate the cell value and modify the errors
-    executeValidateCell(params, headerConfig, errors); // Mutates `errors`
+  // Iterate over each row in the worksheet and execute the row validators
+  forEachCSVRow(worksheet, config, (rowParams, rowValidators) => {
+    // Execute the row validators and modify the errors
+    rowValidators.forEach((rowValidator) => {
+      executeRowValidator(rowParams, rowValidator, errors);
+      // Update the row state for each row validator
+      updateRowState(rowParams, rows);
```

Comment:
Since this is a `forEach` loop, is there any concern around the `updateRowState` calls conflicting with one another? Though, maybe its safe to require that the validation functions don't read/write to the same properties in the state.

Reply:
Latest version handles this better